File: domain/cauldrons.go
package domain

type BDCauldrons map[ID]Cauldron

type Cauldron struct {
	ID       ID     `json:"id"`
	Name     string `json:"name"`
	Capacity int    `json:"capacity"`
	Magmin   int    `json:"magmin"`
	Translit string `json:"translit"`
}

File: frontend/src/models/cauldrons.ts
export namespace cauldrons {
  export interface BDCauldrons {
    [key: number]: Cauldron
  }

  export class Cauldron {
	id: number;
    name: string;
    capacity: number;
    magmin: number;
    translit: string;

    constructor(source: any = {}) {
      if (typeof source === 'string') source = JSON.parse(source);
      this.id = source["id"];
      this.name = source["name"];
      this.capacity = source["capacity"];
      this.magmin = source["magmin"];
      this.translit = source["translit"];
    }

    get title(): string {
      return `${this.translit} (${this.name}) - ${this.capacity} - ${this.magmin}`
    }
  }
}
